django/template/base.py

```diff
@@ -805,7 +809,7 @@ def __init__(self, var):
             except ValueError:
                 # Otherwise we'll set self.lookups so that resolve() knows we're
                 # dealing with a bonafide variable
-                if var.find(VARIABLE_ATTRIBUTE_SEPARATOR + '_') > -1 or var[0] == '_':
+                if VARIABLE_ATTRIBUTE_SEPARATOR + '_' in var or lastchar == '_':
```

Comment:
```suggestion
                if VARIABLE_ATTRIBUTE_SEPARATOR + '_' in var or first_char == '_':
```

Reply:
Fantastic catch, interesting that no test failed...

Reply:
Weirder and weirder. Some of the CI test suite is failing on `tests.template_tests.test_parser.ParserTests.test_variable_parsing` which is what I'm _now_ seeing locally. But some of the others seem to be finishing fine?
(As I write this, 3 have finished OK, 2 have finished with an error)

Reply:
`first_char` will still be set/cached as `_` rather than whatever `var[0]` now points to - the latter may not be the same value because of the line `var = var[2:-1]` for handling i18n strings.